src/common/java/net/minecraftforge/gradle/common/util/runs/EclipseRunGenerator.java

```diff
@@ -93,7 +97,68 @@ protected Map<String, Document> createRunConfiguration(@Nonnull final Project pr
             }
             javaDocument.appendChild(rootElement);
         }
-        documents.put(runConfig.getTaskName() + ".launch", javaDocument);
+
+        final String configName = (mc.generatesRunFolders() ? runConfig.getFolderName() + " - " : "") + runConfig.getTaskName() + ".launch";
+
+        if (mc.enablesEclipsePrepareRuns()) {
+            final String launchConfigName = project.getName() + " - " + runConfig.getTaskName() + "Slim";
+            documents.put(".eclipse/configurations/" + launchConfigName + ".launch", javaDocument);
+
+            // Create a gradle document
+            final Document gradleDocument = documentBuilder.newDocument();
+            {
+                final Element rootElement = gradleDocument.createElement("launchConfiguration");
+                {
+                    rootElement.setAttribute("type", "org.eclipse.buildship.core.launch.runconfiguration");
+
+                    elementAttribute(gradleDocument, rootElement, "string", "gradle_distribution", "GRADLE_DISTRIBUTION(WRAPPER)");
+                    elementAttribute(gradleDocument, rootElement, "boolean", "offline_mode", project.getGradle().getStartParameter().isOffline());
+                    elementAttribute(gradleDocument, rootElement, "boolean", "show_console_view", "true");
+                    elementAttribute(gradleDocument, rootElement, "boolean", "show_execution_view", "true");
+                    elementAttribute(gradleDocument, rootElement, "string", "working_dir", project.getRootDir());
+                }
+
+                final Element tasks = gradleDocument.createElement("listAttribute");
+                tasks.setAttribute("key", "tasks");
+                {
+                    final Element taskEntry = gradleDocument.createElement("listEntry");
+                    taskEntry.setAttribute("value", project.getTasks().getByName("prepare" + Utils.capitalize(runConfig.getTaskName())).getPath());
+                    tasks.appendChild(taskEntry);
+
+                    if (mc.copiesIDEResources()) {
+                        final Element copyIde = gradleDocument.createElement("listEntry");
+                        copyIde.setAttribute("value", project.getTasks().getByName("copyEclipseResources").getPath());
+                        tasks.appendChild(copyIde);
+                    }
+                }
+                rootElement.appendChild(tasks);
+                gradleDocument.appendChild(rootElement);
+            }
```

Comment:
I really dislike the use of empty blocks for organization.

Reply:
I just followed the style of the existing run config generator codebase.